Odata-docs/webapi-8/fundamentals/query-options.md

```diff
@@ -0,0 +1,568 @@
+                {
+                    "Id": 1005,
+                    "Price": 40,
+                    "Quantity": 15
+                }
+            ]
+        }
+    ]
+}
+```
+
+We can combine various query options.
+In the request below:
+- Select only the `Name` property from `Customer` entity.
+- Expand the related `Orders`.
+- Filter the `Orders` and return only the ones whose `Id` is greater than `1004`.
+- Order the `Customers` by `Name` in descending order.
+
+```http
+http://localhost:6285/odata/Customers?$select=Name&$expand=Orders($filter=Id gt 1004)&$orderby=Name desc
+```
+
+```json
+{
+    "@odata.context": "http://localhost:6285/odata/$metadata#Customers(Name,Orders())",
+    "value": [
+        {
+            "Name": "Customer 3",
+            "Orders": [
+                {
+                    "Id": 1006,
+                    "Price": 15,
+                    "Quantity": 50
+                },
+                {
+                    "Id": 1005,
+                    "Price": 40,
+                    "Quantity": 15
+                }
+            ]
+        },
+        {
+            "Name": "Customer 2",
+            "Orders": []
+        },
+        {
+            "Name": "Customer 1",
+            "Orders": []
+        }
+    ]
+}
+```
+
+### Limiting Query Options
+We can limit the type of query option that can be used while calling a certain API action.
+
+```csharp
+public class CustomersController : ODataController
+{
+    [EnableQuery(AllowedQueryOptions = AllowedQueryOptions.Filter)]
+    public IActionResult Get()
+    {
+        return Ok(Customers);
+    }
+}
+```
+
+In the controller method above, we only allow `$filter` query option. If we use another query option e.g `$select`, we will get an error.
+
+```http
+GET http://localhost:6285/odata/Customers?$select=Name
+```
+
+```json
+{
+    "error": {
+        "code": "",
+        "message": "The query specified in the URI is not valid. Query option 'Select' is not allowed. To allow it, set the 'AllowedQueryOptions' property on EnableQueryAttribute or QueryValidationSettings.",
+        "details": [],
+        "innererror": {}
+    }
+}
+```
+
+We can combine multiple query options as shown below:
+
+```csharp
+[EnableQuery(AllowedQueryOptions = AllowedQueryOptions.Filter | AllowedQueryOptions.OrderBy)]
+public IQueryable<Customer> Get()
+{
+    return Customers.AsQueryable<Customer>();
+}
+```
+
+## Config options vs AllowedQueryOptions
+When registering OData services, we add query options configurations as shown below:
+
+```csharp
+.AddOData(
+    options => options.Select().OrderBy().Expand().Count().SetMaxTop(null)
+)
+```
+
+These are global configurations. If a query option is not enabled here, it cannot be enabled in the controller.
+
+In the above example, we did not enable the $filter query option. If we configure a controller as shown below, the $filter will not be applied since we had not enabled $filter in the global configuration.
```

Comment:
```suggestion
In the above example, we did not enable the `$filter` query option. If we configure a controller as shown below, the `$filter` will not be applied since we had not enabled `$filter` in the global configuration.
```